src/Generators/Microsoft.Gen.AutoClient/Emitter.cs

```diff
@@ -226,7 +226,22 @@ private void GenRestApiMethod(RestApiMethod restApiMethod, RestApiType restApiTy
         OutLn();
         OutOpenBrace();
 
-        var pathSb = new StringBuilder(restApiMethod.Path);
+        var encodedPath = restApiMethod.Path;
+
+        if (encodedPath != null && restApiMethod.FormatParameters.Count > 0)
+        {
+            foreach (var param in restApiMethod.FormatParameters)
+            {
+                var escapedParamName = $"{param}Escaped";
+                OutLn($"var {escapedParamName} = {Uri}.EscapeDataString($\"{{{param}}}\");");
+
+                encodedPath = encodedPath.Replace($"{{{param}}}", $"{{{escapedParamName}}}");
```

Comment:
Can we do the replacement on the `StringBuilder`? This will give us a better performance

Reply:
Updated